src/jit/lsra.cpp

```diff
@@ -5705,6 +5705,245 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
+{
+    assert(physRegRecord->assignedInterval != nullptr);
+    RefPosition* recentAssignedRef = physRegRecord->assignedInterval->recentRefPosition;
+
+    if (recentAssignedRef != nullptr)
+    {
+        if (recentAssignedRef->nodeLocation == refLocation)
+        {
+            // We can't spill a register that's being used at the current location
+            return false;
+        }
+
+        // If the current position has the candidate register marked to be delayed,
+        // check if the previous location is using this register, if that's the case we have to skip
+        // since we can't spill this register.
+        if (recentAssignedRef->delayRegFree && (refLocation == recentAssignedRef->nodeLocation + 1))
+        {
+            return false;
+        }
+
+        // We don't prefer to spill a register if the weight of recentAssignedRef > weight
+        // of the spill candidate found so far.  We would consider spilling a greater weight
+        // ref position only if the refPosition being allocated must need a reg.
+        *recentAssignedRefWeight = getWeight(recentAssignedRef);
+        if (*recentAssignedRefWeight > farthestRefPosWeight)
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+#ifdef _TARGET_ARM_
+bool LinearScan::canSpillDoubleReg(RegRecord*   physRegRecord,
+                                   LsraLocation refLocation,
+                                   unsigned*    recentAssignedRefWeight,
+                                   unsigned     farthestRefPosWeight)
+{
+    bool     retVal  = true;
+    unsigned weight  = BB_ZERO_WEIGHT;
+    unsigned weight2 = BB_ZERO_WEIGHT;
+
+    RegRecord* physRegRecord2 = findAnotherHalfRegRec(physRegRecord);
+
+    if (physRegRecord->assignedInterval != nullptr)
+        retVal &= canSpillReg(physRegRecord, refLocation, &weight, farthestRefPosWeight);
+
+    if (physRegRecord2->assignedInterval != nullptr)
+        retVal &= canSpillReg(physRegRecord2, refLocation, &weight2, farthestRefPosWeight);
+
+    if (!(weight == BB_ZERO_WEIGHT && weight2 == BB_ZERO_WEIGHT))
+    {
+        // weight and/or weight2 have been updated.
+        *recentAssignedRefWeight = (weight > weight2) ? weight : weight2;
+    }
+
+    return retVal;
+}
+#endif
+
+//----------------------------------------------------------------------------
+// checkActiveInterval: Test activness of an interval
+//                      and check assertions if the interval is not active
+//
+// Arguments:
+//    interval    - An interval to be tested
+//    refLocation - Location where the interval is being tested
+//
+// Return Value:
+//    True - iff the interval is active
+//    False - otherwise
+//
+// Note: This helper is designed to be used only from checkActiveIntervals()
+//
+bool LinearScan::checkActiveInterval(Interval* interval, LsraLocation refLocation)
+{
+    RefPosition* recentAssignedRef = interval->recentRefPosition;
```

Comment:
Nit: you can sink this load into the `if` statement, as we only need the `recentAssignedRef` if the interval is not active.